### PR TITLE
Fix commission checks plus stake movement limits

### PR DESCRIPTION
**Pull request by alex-at-planetariummusic**

### CODE REVIEW

File: bot/src/stake_pool_v0.rs [hunk 1]
```diff
@@ -24,6 +25,9 @@ const MIN_STAKE_DELEGATION: u64 = 1000000000;
 // Delegation rent amount. Need
 const DELEGATION_RENT: u64 = 2282880;
 
+// Do not move more than this amount of stake per epoch
+const MAX_SOL_MOVEMENT_PER_EPOCH: u64 = 5_000_000;
```
Comment (t-nelson) [inline bot/src/stake_pool_v0.rs:29]: wdyt of making this a percentage and resolving the actual number of tokens at runtime?  this value isn't going to work for all implementations.

NOTE: _highly_ recommend staying in integers. that is, `MAX_SOL_PCT_INT * total_staked / 100` as opposed to `MAX_SOL_PCT_FLT * total_staked`

File: bot/src/stake_pool_v0.rs [hunk 2]
```diff
@@ -655,6 +662,16 @@ where
...
             })?;
 
+        let desired_balance = match validator_stake.stake_state {
+            ValidatorStakeState::None => MIN_STAKE_ACCOUNT_BALANCE,
+            ValidatorStakeState::Baseline => baseline_stake_amount,
+            ValidatorStakeState::Bonus => bonus_stake_amount,
+        };
+        if desired_balance < balance {
+            total_desired_stake_removal =
+                total_desired_stake_removal.wrapping_add(desired_balance.wrapping_sub(balance));
```
Comment (t-nelson) [inline bot/src/stake_pool_v0.rs:672]: almost positive we don't want to wrap here. saturation is safer

**Review by t-nelson — COMMENTED** [review]
left a couple comments.  ping me when it's in final form

File: bot/src/stake_pool_v0.rs (hunk 1 above)
Comment (alex-at-planetariummusic) [reply bot/src/stake_pool_v0.rs:29]: > wdyt of making this a percentage and resolving the actual number of tokens at runtime? this value isn't going to work for all implementations.

Yeah, this is at best a stopgap. Wasn't sure offhand how to get the actual number of tokens; could you point me in the right direction?

**Pull request closed without merging**